File: 2023/go/day1/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
	"strconv"
)

func main() {
	f, err := os.Open("./input.txt")
	if err != nil {
		panic(err)
	}
	defer f.Close()
	scan := bufio.NewScanner(f)
	var total = 0
	for scan.Scan() {
		line := scan.Text()
		d, err := regexp.Compile(`\d`)
		if err != nil {
			panic(err)
		}
		digits := d.FindAllString(line, -1)
		num := digits[0] + digits[len(digits)-1]
		toSum, err := strconv.Atoi(num)
		if err != nil {
			panic(err)
		}
		total = total + toSum
	}
	fmt.Printf("The total is %d \n", total)
}

File: 2023/rust/day1/src/main.rs
use regex::Regex;
use std::fs::read_to_string;

fn main() {
    let mut total: i32 = 0;
    let reg = Regex::new(r"\d").unwrap();
    for line in read_to_string("./input.txt").unwrap().lines() {
        let digits: Vec<&str> = reg.find_iter(line).map(|m| m.as_str()).collect();
        let num = format!("{}{}", digits[0], digits[digits.len() - 1]);
        total = total + num.parse::<i32>().unwrap();
    }
    println!("Your total is {}", total);
}
